cmd/stats.go

```diff
@@ -0,0 +1,381 @@
+	case v < 0.0:
+		logger.Fatalf("invalid value %f", v)
+	case v == 0.0:
+		ret, color = " 0.0", WHITE
+	case v < 30.0:
+		ret, color = fmt.Sprintf("%4.1f", v), GREEN
+	case v < 100.0:
+		ret, color = fmt.Sprintf("%4.1f", v), YELLOW
+	default:
+		ret, color = fmt.Sprintf("%4.f", v), RED
+	}
+	return w.colorize(ret, color, dark, false) +
+		w.colorize("%", BLACK, false, false)
+}
+
+func (w *statsWatcher) printDiff(left, right map[string]float64, dark bool) {
+	if !w.tty && dark {
+		return
+	}
+	values := make([]string, len(w.sections))
+	for i, s := range w.sections {
+		vals := make([]string, 0, len(s.items))
+		for _, it := range s.items {
+			switch it.typ & 0xF0 {
+			case metricGauge: // currently must be metricByte
+				vals = append(vals, w.formatU64(right[it.name], dark, true))
+			case metricCounter:
+				v := (right[it.name] - left[it.name])
+				if !dark {
+					v /= float64(w.interval)
+				}
+				if it.typ&metricByte != 0 {
+					vals = append(vals, w.formatU64(v, dark, true))
+				} else if it.typ&metricCPU != 0 {
+					vals = append(vals, w.formatCPU(v, dark))
+				} else { // metricCount
+					vals = append(vals, w.formatU64(v, dark, false))
+				}
+			case metricHist: // metricTime
+				count := right[it.name+"_total"] - left[it.name+"_total"]
+				var avg float64
+				if count != 0.0 {
+					cost := right[it.name+"_sum"] - left[it.name+"_sum"]
+					if it.typ&metricTime != 0 {
+						cost *= 1000 // s -> ms
+					}
+					avg = cost / count
+				}
+				if !dark {
+					count /= float64(w.interval)
+				}
+				vals = append(vals, w.formatU64(count, dark, false), w.formatTime(avg, dark))
+			}
+		}
+		values[i] = strings.Join(vals, " ")
+	}
+	if w.tty && dark {
+		fmt.Printf("%s\r", strings.Join(values, w.colorize("|", BLUE, true, false)))
+	} else {
+		fmt.Printf("%s\n", strings.Join(values, w.colorize("|", BLUE, true, false)))
+	}
+}
+
+func stats(ctx *cli.Context) error {
+	setLoggerLevel(ctx)
+	if ctx.Args().Len() < 1 {
+		logger.Fatalln("mount point must be provided")
+	}
+	mp := ctx.Args().First()
+	inode, err := utils.GetFileInode(mp)
+	if err != nil {
+		logger.Fatalf("lookup inode for %s: %s", mp, err)
+	}
+	if inode != 1 {
+		logger.Fatalf("path %s is not a mount point", mp)
+	}
+
+	watcher := &statsWatcher{
+		tty:      !ctx.Bool("nocolor") && isatty.IsTerminal(os.Stdout.Fd()),
+		interval: ctx.Uint("interval"),
+		path:     path.Join(mp, ".stats"),
+	}
+	watcher.buildSchema(ctx.String("schema"), ctx.Uint("verbosity"))
+	watcher.formatHeader()
+
+	var tick uint
+	var start, last, current map[string]float64
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+	current = readStats(watcher.path)
+	start = current
+	last = current
+	for {
+		if tick%(watcher.interval*30) == 0 {
+			fmt.Println(watcher.header)
+		}
+		if tick%watcher.interval == 0 {
+			watcher.printDiff(start, current, false)
+			start = current
+		} else {
+			watcher.printDiff(last, current, true)
+		}
+		last = current
+		tick++
+		<-ticker.C
```

Comment:
nit:  `time.Sleep(time.Second)` works the same.

Reply:
use a standalone ticker will be more accurate in case `readStats` or `printDiff` takes considerable time

Reply:
Sounds good